server/controllers/incidents.js

```diff
@@ -5,104 +5,149 @@ const Level = require("../models").Levels;
-  attributes: ['status']
-}, {
-  model: Location,
-  attributes: ['name', 'centre', 'country']
-}, {
-  model: User, 
-  as: 'assignees',
-  userAttributes,
-  through: {
-    attributes: ['assignedRole']
-  }
-},
-{
-  model: User, 
-  as: 'reporter',
-  userAttributes,
-  through: {
-    attributes: []
+let includes = [
+  {
+    model: Level,
+    attributes: ["name"]
+  },
+  {
+    model: Status,
+    attributes: ["status"]
+  },
+  {
+    model: Location,
+    attributes: ["name", "centre", "country"]
+  },
+  {
+    model: User,
+    as: "assignees",
+    userAttributes,
+    through: {
+      attributes: ["assignedRole"]
+    }
+  },
+  {
+    model: User,
+    as: "reporter",
+    userAttributes,
+    through: {
+      attributes: []
+    }
+  },
+  {
+    model: User,
+    as: "witnesses",
+    userAttributes,
+    through: {
+      attributes: []
+    }
   }
-}]
+];
 
 // mapping Assignees
-const mapAssignees = (incident) => {
-return incident.map(oneIncident => {
-  oneIncident.dataValues.assignedRole = oneIncident.dataValues.assigneeIncidents.assignedRole;
+const mapAssignees = incident => {
+  return incident.map(oneIncident => {
+    oneIncident.dataValues.assignedRole =
+      oneIncident.dataValues.assigneeIncidents.assignedRole;
     delete oneIncident.dataValues.assigneeIncidents;
-    return oneIncident
-    });
-  }
+    return oneIncident;
+  });
+};
 
 module.exports = {
   // create an incident
   create(req, res) {
     let locationPromise;
-    name = req.body.name,
-    centre = req.body.centre,
-    country = req.body.country
-    return LocationService.create(name, centre, country).then(location => {
-      return location.dataValues.id;
-    }).then(locationId=> {
-      return Incident.create({
-        description: req.body.description,
-        subject: req.body.subject,
-        dateOccurred: req.body.dateOccurred,
-        statusId: req.body.statusId || 1,
-        locationId,
-        levelId: req.body.levelId
-      });
-    }).then(incident=> {
-      return User.findById(req.body.userId).then(user => {
-        return incident.addReporter(user);
-      }).then(()=> {
-        return incident;
+    (name = req.body.name),
+      (centre = req.body.centre),
+      (country = req.body.country);
+    return LocationService.create(name, centre, country)
+      .then(location => {
+        return location.dataValues.id;
+      })
+      .then(locationId => {
+        return Incident.create({
+          description: req.body.description,
+          subject: req.body.subject,
+          dateOccurred: req.body.dateOccurred,
+          statusId: req.body.statusId || 1,
+          locationId,
+          levelId: req.body.levelId
+        });
+      })
+      .then(incident => {
+        return User.findById(req.body.userId)
+          .then(user => {
+            incident.addReporter(user);
+            return incident;
+          })
+          .then(incident => {
+            let userId = req.body.id;
```

Comment:
Can you sync with whoever is sending this data so that we don't carry out map on a variable named in singular like `id`. Something like `ids` and assign it appropriately. It doesn't make sense calling map on a variable called `userId`